### ISSUE TITLE
Crashed when trying to import "gnu-dummy" S2K key pairs

### ISSUE THREAD

**Issue by h12w** [mssun/passforios#61]
### Environment

iOS Version: 10.2.1 (14D27)
Pass for iOS: 0.2 (24)

### Steps

1. Export PGP public and private key (4096 RSA key generated by GnuPG) in ASCII-Armor format using the instructions:
```
gpg --export -a <keyname>
gpg --export-secret-keys -a <keyname>
```

2. Paste the key into the two edit controls of "PGP Key" window

3. Press "Save" in the upper right corner

4. Enter passphrase in the "PassPhrase" dialog, and press "OK", then press "Save"

### Expected Result

The PGP key is saved

### Got Result

The program crashed

### Crash Log

[pass_ios_crash_log.txt](https://github.com/mssun/passforios/files/847013/pass_ios_crash_log.txt)

**Comment by yishilin14** [comment]
@h12w Thank you for your report. I will use a 4096 RSA key to test later.

**Comment by h12w** [comment]
It may not related to 4096 bit RSA key. One of my colleague also uses 4096 key but succeeded.

**Comment by yishilin14** [comment]
I see, thanks. We are polishing the codes now. Sometime the app crashes because of some other (non-obvious) reasons.

I have some questions.
- Are you able to crash the app every time you save your keys (e.g., after reinstalling the app)?
- Would you please take a look at your pasted keys and see whether they look normal? Just in case the whatever method helping you get your keys to your phone changes something (e.g, transform --- to Chinese —).

I just tried this: I used a randomly typed key and a randomly typed passphrase, saving them crashes the app... Not sure whether your app also crashed because of the invalid key... (I will fix the bug I found...)

**Comment by h12w** [comment]
* It crashes every time I save my keys, even if I reinstalled the app.
* My keys look normal, like below:

```
-----BEGIN PGP PUBLIC KEY BLOCK-----
Version: GnuPG v1

...
...
-----END PGP PUBLIC KEY BLOCK-----
```

and 

```
-----BEGIN PGP PRIVATE KEY BLOCK-----
Version: GnuPG v1

...
...
-----END PGP PRIVATE KEY BLOCK-----
```

**Comment by yishilin14** [comment]
Thank you so much. Let me try to work around this issue, at least provide some readable "error messages".

**Comment by mssun** [comment]
I'm closing this issue. Should be fixed in the recent commits: d338e725d5107774165ad71e4c799b809c18c97b

I plan release a new version this weekend. Thanks.

**Comment by h12w** [comment]
I have tested against the new version, a similar crash problem still exists. Here is the report:

### Env

iOS Version: 10.2.1 (14D27)
Pass for iOS: 0.2.1 (1)

### Steps

same as above

### Expected Result

The PGP key is saved

### Got Result

The app crashed, and the app **crashes every time I click on the app icon ever since**.

### Crash Log

[pass_ios_crash_log_0.2.1_1.txt](https://github.com/mssun/passforios/files/852185/pass_ios_crash_log_0.2.1_1.txt)

**Comment by mssun** [comment]
I couldn't debug the crash without your keys.

Can you generate a test key pair and paste them here?

**Comment by h12w** [comment]
I generated a new pair of keys that can crash the app:

[testpassios.prv.txt](https://github.com/mssun/passforios/files/852353/testpassios.prv.txt)
[testpassios.pub.txt](https://github.com/mssun/passforios/files/852354/testpassios.pub.txt)

And here is my way to generate them: [Password Management](http://h12.me/article/password-management), following the steps in section "Preparing PGP Keys", "Backup PGP keys" and "Hide PGP Master Private Key".

**Comment by mssun** [comment]
Thank you. Now, I can reproduce the bug. Could be caused by a bug in an upstream library. I'm debugging it.

**Comment by mssun** [comment]
Hi, I did some research these days.

The problem is caused by the "Hide PGP Master Private Key" step. This step will generate a "gnu-dummy S2K" which is new and not supported by our upstream library (ObjectivePGP). I have reported the issue in https://github.com/krzyzanowskim/ObjectivePGP/issues/53.

For now, if you want to use the app, please do not hide PGP master private key. Thanks.

**Comment by mssun** [comment]
@h12w Have you tried my solution? Any update? Thanks.

This is an upstream issue, I tried my best to ping the author of ObjectivePGP for fixing. Thanks for your patience.

**Comment by h12w** [comment]
Thanks. It works after I restore my local key pairs without hiding the master key and do the import again. I will submit reports if I encounter any other issues.

**Comment by mssun** [comment]
@h12w The crash has been fixed in ObjectivePGP. Can you try again in the latest TestFlight release or master branch.

**Comment by yishilin14** [comment]
@mssun It seems like the pair of keys provided by @h12w still crash the app.

**Comment by mssun** [comment]
Should be ok this time. Please try. @yishilin14

**Comment by yishilin14** [comment]
Fixed in TestFlight release 0.2.6 (32).
